### PR TITLE
Use Tuples for useState

### PR DESCRIPTION
**Pull request by mcordova47**

### CODE REVIEW

**Review by fsoikin — APPROVED** [review]
Wow, I didn't know operators worked as patterns. Nice job!

File: examples/src/Examples/UseLocalStorage.purs [hunk 1]
```diff
@@ -32,17 +34,14 @@ view = withHooks do
...
     case v of
       Just v' -> setState v'
       Nothing -> setItem key defaultValue =<< localStorage =<< window
-  pure
-    { value: state
-    , setValue: \v -> do
-        setState v
-        setItem key v =<< localStorage =<< window
-    }
+  pure $ Tuple state \v -> do
```
Comment (fsoikin) [inline examples/src/Examples/UseLocalStorage.purs:45]: I think this should work with the operator too, right?

```suggestion
  pure $ state /\ \v -> do
```

Comment (mcordova47) [reply examples/src/Examples/UseLocalStorage.purs:45]: Yeah, I wasn’t sure which was more readable. I could even use it in the type signature if we want.

File: examples/src/Examples/UseLocalStorage.purs [hunk 2]
```diff
@@ -34,14 +33,14 @@ view = withHooks do
       ]
     ]
 
-useLocalStorage :: HookName -> String -> String -> Hook (Tuple String (Dispatch String))
+useLocalStorage :: HookName -> String -> String -> Hook (String /\ (Dispatch String))
```
Comment (fsoikin) [inline examples/src/Examples/UseLocalStorage.purs:36]:
```suggestion
useLocalStorage :: HookName -> String -> String -> Hook (String /\ Dispatch String)
```

File: src/Elmish/Hooks/UseState.purs [hunk 3]
```diff
@@ -5,14 +5,15 @@ module Elmish.Hooks.UseState
...
+import Data.Tuple (curry)
+import Data.Tuple.Nested (type (/\))
 import Elmish (Dispatch, ReactElement)
 import Elmish.Component (ComponentName, wrapWithLocalState)
 import Elmish.Hooks.Type (Hook, HookName, mkHook)
 
 type Args state =
   { initialState :: state
-  , render :: Tuple state (Dispatch state) -> ReactElement
+  , render :: state /\ (Dispatch state) -> ReactElement
```
Comment (fsoikin) [inline src/Elmish/Hooks/UseState.purs:16]:
```suggestion
  , render :: state /\ Dispatch state -> ReactElement
```

File: src/Elmish/Hooks/UseState.purs [hunk 4]
```diff
@@ -24,6 +25,6 @@ useState' name = wrapWithLocalState name \{ initialState, render } ->
 
 -- | The `useState` hook takes an initial state and its callback has access to
 -- | the current state and a setter for the state.
-useState :: forall state. HookName -> state -> Hook (Tuple state (Dispatch state))
+useState :: forall state. HookName -> state -> Hook (state /\ (Dispatch state))
```
Comment (fsoikin) [inline src/Elmish/Hooks/UseState.purs:28]:
```suggestion
useState :: forall state. HookName -> state -> Hook (state /\ Dispatch state)
```

**Pull request merged**